src/sparkle/application/__init__.py

```diff
@@ -1,68 +1,132 @@
 import abc
+import os
+from typing import Any, Dict, List
 from pyspark.sql import SparkSession, DataFrame
 from pyspark.storagelevel import StorageLevel
-from sparkle.config import Config
+from pyspark import SparkConf, SparkContext
+from pyspark.sql import SparkSession
+from pyspark.context import SparkContext
+from sparkle.config import Config, ExecutionEnvironment
 from sparkle.writer import Writer
 from sparkle.reader import Reader
+from sparkle.utils.logger import logger
 
-PROCESS_TIME_COLUMN = "process_time"
 
+_SPARK_EXTENSIONS = [
+    "org.apache.iceberg.spark.extensions.IcebergSparkSessionExtensions",
+]
+_SPARK_PACKAGES = [
+    "org.apache.iceberg:iceberg-spark-runtime-3.3_2.12:1.3.1",
+    "org.apache.spark:spark-sql-kafka-0-10_2.12:3.3.0",
+    "org.apache.spark:spark-avro_2.12:3.3.0",
+]
 
-class Sparkle(abc.ABC):
-    """Base class for Spark applications.
-
-    This is an abstract base class for defining Spark applications.
-    It provides a common interface for processing data and writing the
-    results using a set of configurable writers.
 
-    Attributes:
-        spark_session (SparkSession): The Spark session used for
-          executing Spark jobs.
-        config (Config): Configuration object containing
-          application-specific settings.
-        sources (Dict[str, Source]): Dictionary of data sources used
-          in the application, keyed by source name.
-        writers (List[Writer]): List of Writer objects used for
-          writing the processed data to various destinations.
-
-    """
+class Sparkle(abc.ABC):
+    """Base class for Spark applications."""
 
     def __init__(
         self,
-        spark_session: SparkSession,
+        env: ExecutionEnvironment,
```

Comment:
@kkiani @shahinism For AWS case, it some how needs to know if it's running in Glue or not because of this line:
```
GlueContext(SparkContext.getOrCreate(conf=spark_conf)).spark_session
```

Let me know if you have any idea about it.